[Pytorch Delegated Backend] Save operator name and function name in

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58873 [Pytorch backend delegation] Preprocess to accept
* #57481 [Pytorch Delegated Backend] Save function name in debug info
* **#57441 [Pytorch Delegated Backend] Save operator name and function name in**

debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

kimishpatel committed:
[Pytorch Delegated Backend] Save operator name and function name in
debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

raziel commented on torch/csrc/jit/serialization/export_module.cpp:
```diff
@@ -341,12 +341,12 @@ SourceRangeRecords getBackendSourceRanges(const Module& m) {
-        auto cs_ptr = it.second.second;
+        auto& source_range = std::get<kDebugInfoTupleSourceRangeIndex>(it.second);
+        sr_records.emplace_back(std::numeric_limits<size_t>::max(), source_range);
+        auto cs_ptr = std::get<kDebugInfoTupleInlinedCSIndex>(it.second);
```
const ref

raziel reviewed (approved):
Some comments to address first.

Thanks

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Save operator name and function name in"


debug info

Summary:
Previous diffs did not save operator name in debug info. For delegated
backends that only idenfity op for profiling with debug handle, operator
name should be stores as well.
Furthermore to complete debug informaton also serialize function name.

Test Plan:
Existing lite interpreter and backend tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28144581](https://our.internmc.facebook.com/intern/diff/D28144581)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28144581).
